crates/sim2h/src/job.rs

```diff
@@ -0,0 +1,20 @@
+/// typedef for code clarity
+pub(crate) type JobContinue = bool;
+
+/// an item that can be executed via thread pool
+pub(crate) trait Job: 'static + Send {
```

Comment:
Where does the `'static` constraint come in from?

Reply:
Makes things sooo much easier, because I don't have to specify lifetimes on the trait-object job instances - just means we can't hold non-static references in the closures / structs for the jobs, which I'm totally cool with.